specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,577 @@
+  /// Listen to stop stream request once the stream started.
+  /// It is called when user stop all streaming requests from
+  /// the renderers (Javascript) or the host calls the Stop API. The renderer
+  /// can stream again by calling the streaming request API.
+  /// The renderer cleared all registered TextureBuffers before sending
+  /// the stop request event so that the callback of the next start request
+  /// should register the textures again.
+  /// The event is triggered when all requests for given stream id closed
+  /// by the Javascript, or the host's Stop API call.
+  /// TextureBuffer related API calls after this event will return an error
+  /// of HRESULT_FROM_WIN32(ERROR_INVALID_STATE).
+  HRESULT add_Stopped(
+      [in] ICoreWebView2StagingTextureStreamStoppedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for Stopped event.
+  HRESULT remove_Stopped(
+      [in] EventRegistrationToken token);
+  /// Creates TextureBuffer that will be referenced by the host and the browser.
+  /// By using the TextureBuffer mechanism, the host does not have to
+  /// send respective texture to the renderer, instead it notifies it
+  /// with internal TextureBuffer id, which is the identity of the TextureBuffer.
+  /// The TextureBuffer is 2D texture, IDXGIResource, format and will be
+  /// exposed through shared HANDLE or IUnknown type through ICoreWebView2StagingTexture.
+  /// Whenever the host has new texture to write, it should ask
+  /// reusable ICoreWebView2StagingTexture from the GetAvailableBuffer,
+  /// which returns ICoreWebView2StagingTexture.
+  /// If the GetAvailableBuffer returns an error, then the host calls the
+  /// CreateBuffer to allocate new TextureBuffer.
+  /// The API also registers created shared handle to the browser once it
+  /// created the resource.
+
+  /// Unit for width and height is texture unit (in texels).
+  /// 'https://learn.microsoft.com/en-us/windows/win32/api/d3d11ns-d3d11-d3d11_texture2d_desc'
+  HRESULT CreateBuffer(
+    [in] UINT32 widthInTexels,
+    [in] UINT32 heightInTexels,
+    [out, retval] ICoreWebView2StagingTexture** buffer);
+  /// Returns reuseable TextureBuffer for video frame rendering.
+  /// Once the renderer finishes rendering of TextureBuffer's video frame, which
+  /// was requested by Present, the renderer informs the host so that it can
+  /// be reused. The host has to create new TextureBuffer with CreateBuffer
+  /// if the API return an error HRESULT_FROM_WIN32(ERROR_NO_MORE_ITEMS).
+  HRESULT GetAvailableBuffer([out, retval] ICoreWebView2StagingTexture** buffer);
+  /// Removes TextureBuffer when the host removes the backed 2D texture.
+  /// The host can save the resources by deleting 2D textures
+  /// when it changes the frame sizes. The API will send a message
+  /// to the browser where it will remove TextureBuffer.
+  HRESULT CloseBuffer([in] ICoreWebView2StagingTexture* buffer);

```

Comment:
Add a note to spec to discuss with reviewers if this is better as
TextureStream.CloseTexture
or
Texture.Close

since it also involves the TextureStream managing sharing resources with the browser.